server/eclipse-project/src/main/java/us/freeandfair/corla/persistence/Persistence.java

```diff
@@ -359,11 +371,16 @@ public static boolean beginTransaction()
     checkForDatabase();
 
     boolean result = true;
-    final Session session = currentSession();
+    Session session = currentSession();
+    
     if (isTransactionActive()) {
       result = false;
     } else if (canTransactionRollback()) {
       rollbackTransaction();
+      // session is explicitly closed by rollback
+      // interesting note: isOpen() on the session _still returns true_ even though 
```

Comment:
Does `isSessionOpen()` return `true` in this case, too?

Reply:
Yes, it does, but I still think it's a worthwhile thing to have for potential other uses. I think I should also report it to the Hibernate people, when I get a chance.